### PR TITLE
[Fix Bug] ResNet50 pretrained weights being ignored

### LINKED ISSUE #51: ResNet50 pretrained is ignored
## Bug
Hey,

The args `use_pretrain` and `pretrain_path` a currently ignored for ResNet50 and an untrained model is always returned.
I prepared a small PR to fix this.

Thanks a lot!

### PR DESCRIPTION
**Pull request by thomasbohm**
## What does this PR do?

Fixes #51 

Currently, pretrained weights are ignored for ResNet50.
I am using torchvision's ImageNet weights since there are no USB resnet50 weights yet.

To try this out, e.g. use this as `pretrained_path`: 
[https://download.pytorch.org/models/resnet50-11ad3fa6.pth](https://download.pytorch.org/models/resnet50-11ad3fa6.pth) 

Then, modify `semilearn/nets/utils.py:21` to:
`orig_state_dict = checkpoint['model'] if 'model' in checkpoint.keys() else checkpoint`

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you list all the **breaking changes** introduced by this pull request?

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

**Review by Hhhhhhao — APPROVED** [review]

**Comment by Hhhhhhao** [comment]
Thanks for the pull request! It has been merged.
